src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Tables/Instance.sql

```diff
@@ -45,98 +47,84 @@ INCLUDE
 )
 WITH (DATA_COMPRESSION = PAGE)
 
--- Used in WADO and QIDO, putting PartitionKey last allows us to query across partitions in the future.
-CREATE NONCLUSTERED INDEX IX_Instance_StudyInstanceUid_Status_PartitionKey on dbo.Instance
+-- Used in WADO
+CREATE NONCLUSTERED INDEX IX_Instance_PartitionKey_Status_StudyInstanceUid_SeriesInstanceUid_SopInstanceUid on dbo.Instance
 (
-    StudyInstanceUid,
+    PartitionKey,
     Status,
-    PartitionKey    
-)
-INCLUDE
-(
-    SeriesInstanceUid,
-    SopInstanceUid,
-    Watermark,
-    TransferSyntaxUid
-)
-WITH (DATA_COMPRESSION = PAGE)
-
--- Used in WADO and QIDO, putting PartitionKey last allows us to query across partitions in the future.
-CREATE NONCLUSTERED INDEX IX_Instance_StudyInstanceUid_SeriesInstanceUid_Status_PartitionKey on dbo.Instance
-(
     StudyInstanceUid,
     SeriesInstanceUid,
-    Status,
-    PartitionKey    
+    SopInstanceUid
 )
 INCLUDE
 (
-    SopInstanceUid,
     Watermark,
-    TransferSyntaxUid
+    TransferSyntaxUid,
+    HasFrameMetadata
 )
 WITH (DATA_COMPRESSION = PAGE)
 
--- Used in WADO and QIDO, putting PartitionKey last allows us to query across partitions in the future.
-CREATE NONCLUSTERED INDEX IX_Instance_SopInstanceUid_Status_PartitionKey on dbo.Instance
+-- Used in GetInstancesByWatermarkRange
+CREATE UNIQUE NONCLUSTERED INDEX IX_Instance_Watermark_Status on dbo.Instance
 (
-    SopInstanceUid,
-    Status,
-    PartitionKey    
+    Watermark,
+    Status
 )
 INCLUDE
 (
+    PartitionKey,
     StudyInstanceUid,
     SeriesInstanceUid,
-    Watermark
+    SopInstanceUid
 )
 WITH (DATA_COMPRESSION = PAGE)
 
--- Used in GetInstancesByWatermarkRange
-CREATE UNIQUE NONCLUSTERED INDEX IX_Instance_Watermark_Status on dbo.Instance
+-- QIDO filtering
+CREATE UNIQUE NONCLUSTERED INDEX IX_Instance_PartitionKey_SopInstanceUid ON dbo.Instance
```

Comment:
Shouldn't the partitionKey be at the end for QIDO?

Reply:
Same as above